### DISCUSSION TITLE
Ideas on CPU load (reduction) optimization?

### DISCUSSION THREAD

**Discussion by RagnarSTS** [Security-Onion-Solutions/securityonion#7842] [Unsupported Versions] [answered]
On normal server class hardware CPU load isn't a problem.  Normal increased capabilities overall and ability to add nodes makes it no issue.  Home lab is a somewhat different category.  I'm curious if there are methods to reduce general tasks in a low throughput environment to keep cpu cycles down a bit.  

- Sensor node seems to be ideal already; appropriate cpu cycles for the work it is doing.

- IDH node is maybe capable of a tiny trim as it really doesn't process anything most of the time but seems to just generate cycles for no reason.

- The master/search node is where I'm really curious if tweaks can be made.  Elastic will simply run along on it with a java node pretty much all day long.  Traffic isn't exactly high; and when I do throw a lot of traffic at the sensors the processing doesn't really spike up much at all.  It really seems to be just keeping itself busy on never ending background processes.  

Not sure where to start poking at the system in general: Normal cron jobs or the mysticism of the saltstack?  
Appreciate any thoughts on the matter.

**Comment by dougburks** [comment, marked as answer]
> * IDH node is maybe capable of a tiny trim as it really doesn't process anything most of the time but seems to just generate cycles for no reason.

Do you have Wazuh enabled? If so, that is most likely what is generating the cycles.

> * The master/search node is where I'm really curious if tweaks can be made.  Elastic will simply run along on it with a java node pretty much all day long.  Traffic isn't exactly high; and when I do throw a lot of traffic at the sensors the processing doesn't really spike up much at all.  It really seems to be just keeping itself busy on never ending background processes.

Do you have Playbook enabled? If so, that is going to run queries on a regular basis.

You might also be seeing periodic Curator jobs.

You might try installing an Import node and comparing that to your master/search node as Import would be more of a minimum Elasticsearch instance.

**Reply by RagnarSTS** [reply]
Concerning Wazuh, I know I turned it off once before but can't seem to find any references in the discussion section or docs about how to repeat the task.  Most likely this was an oversight on my part during one of the last reloads on the home lab.  

Playbook is enabled; but only ~30ish plays enabled.  I could potentially turn some off as they might be extraneous.

I have noted that an import node is pretty quiet on another lab setup.

**Reply by dougburks** [reply]
> Concerning Wazuh, I know I turned it off once before but can't seem to find any references in the discussion section or docs about how to repeat the task. Most likely this was an oversight on my part during one of the last reloads on the home lab.

Wazuh can be disabled during Setup:
![image](https://user-images.githubusercontent.com/1659467/165757979-bc6f98b0-5dd5-44c4-a2d0-0bc2b2059a24.png)

Disabling Wazuh after Setup is not really supported. You might be able to disable it in the global pillar, remove it from so-status.conf, and disable the agent service. But if you decide to try that, I would recommend trying it on a test box.
